api/preprints/views.py

```diff
@@ -227,3 +232,145 @@ class PreprintContributorsList(NodeContributorsList, PreprintMixin):
+        =========================================================================
+        name           string             title of the institution
+        id             string             unique identifier in the OSF
+        logo_path      string             a path to the institution's static logo
+
+
+
+    #This Request/Response
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_PREPRINTS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    serializer_class = PreprintProviderSerializer
+
+    model = PreprintProvider
+    view_category = 'preprints'
+    view_name = 'preprint-preprint_providers'
+
+    def get_queryset(self):
+        node = self.get_node()
+        return node.preprint_providers or []
+
+
+class PreprintToPreprintProviderRelationship(JSONAPIBaseView, generics.RetrieveUpdateDestroyAPIView, generics.CreateAPIView, PreprintMixin):
+    """ Relationship Endpoint for Preprint -> PreprintProvider
+
+    Used to set preprint_provider of a preprint to a PreprintProvider
+
+    ##Actions
+
+    ###Get
+
+        Method:        GET
+        URL:           /links/self
+        Query Params:  <none>
+        Success:       200
+
+    ###Create
+
+        Method:        POST
+        URL:           /links/self
+        Query Params:  <none>
+        Body (JSON):   {
+                         "data": [
+                             {
+                               "type": "preprint_providers",   # required
+                               "id": <provider_id>   # required
+                             }
+                         ]
+                       }
+        Success:       201
+
+    ###Update
+
+        Method:        PUT || PATCH
+        URL:           /links/self
+        Query Params:  <none>
+        Body (JSON):   {
+                         "data": [{
+                           "type": "preprint_providers",   # required
+                           "id": <provider_id>   # required
+                         }]
+                       }
+        Success:       200
+
+        This will delete all preprint_providers not listed, meaning a data: [] payload
+        does the same as a DELETE with all the preprint_providers.
+
+    ###Destroy
+
+        Method:        DELETE
+        URL:           /links/self
+        Query Params:  <none>
+        Body (JSON):   {
+                         "data": [{
+                           "type": "preprint_providers",   # required
+                           "id": <provider_id>   # required
+                         }]
+                       }
+        Success:       204
+
+    All of these methods require admin permissions in the preprint.
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.ALWAYS_PUBLIC]
```

Comment:
I'm not sure that this should be `ALWAYS_PUBLIC` -- it contains mutable and possibly personally identifiable information. Consider `NODE_PREPRINTS_READ` instead